model/pod/dispatcher/dispatcher.go

```diff
@@ -74,3 +86,201 @@ func (pd *PodDispatcher) UpsertAtomically() (*adb.ChangeInfo, error) {
 	pd.ModificationCount++
 	return change, nil
 }
+
+// AssignNextTask assigns the pod the next available task to run. Returns nil if
+// there's no task available to run.
+func (pd *PodDispatcher) AssignNextTask(env evergreen.Environment, p *pod.Pod) (*task.Task, error) {
+	ctx, cancel := env.Context()
+	defer cancel()
+
+	grip.WarningWhen(len(pd.TaskIDs) > 1, message.Fields{
```

Comment:
Since we don't support task groups for now, I think it's okay to just warn indicating that we're not supposed to test this yet. The container task queue doesn't support task groups anyways, so this is just a reminder to us to not try using task groups yet until we implement it.